src/inject/gradeDist.js

```diff
@@ -5,153 +5,101 @@ function GradeDist(errorHandler) {
         {label: 'C', dataField: 'cPercent'},
         {label: 'D', dataField: 'dPercent'},
         {label: 'F', dataField: 'fPercent'},
-        {label: 'W', dataField: 'wPercent'},
         {label: 'P', dataField: 'pPercent'},
-        {label: 'NP', dataField: 'npPercent'},
-        {label: 'GPA', dataField: 'gpa'}
+        {label: 'NP', dataField: 'npPercent'}
     ], 'Grade Distribution data does not exist for this professor.', errorHandler);
 }
 
 GradeDist.prototype = Object.create(DataSection.prototype);
 GradeDist.prototype.constructor = GradeDist;
 
 GradeDist.prototype.getNewData = function (teacher, course, callback) {
-    var url = 'http://asucsd.ucsd.edu/gradeDistribution?' +
-        'GradeDistribution%5BTERM_CODE%5D=' +
-        '&GradeDistribution%5BSUBJECT_CODE%5D='+ course.subjectCode +
-        '&GradeDistribution%5BCOURSE_CODE%5D=' + course.courseCode +
-        '&GradeDistribution%5BCRSE_TITLE%5D=' +
-        '&GradeDistribution%5BINSTRUCTOR%5D=' + encodeURIComponent(teacher.nomiddle) +
-        '&GradeDistribution%5BGPA%5D=' +
-        '&GradeDistribution_page=1' +
-        '&ajax=gradedistribution-grid';
+    // function that has to be here
+    this.fetchHTMLHttp("", function (page) {
```

Comment:
The reason it didn't work when you took it out is because the function uses a callback. That means that the function calling it isn't expecting it to return a value, but instead is expecting it to do `return callback();` What should happen is that after you set `this.data` then you do `return callback();`. That way, you can remove this fetchHTMLHttp

Here is a short intro to the concept of callbacks: http://dreamerslab.com/blog/en/javascript-callbacks/